src/meta/src/barrier/info.rs

```diff
@@ -257,6 +259,92 @@ impl InflightDatabaseInfo {
             }
         }
     }
+
+    pub(super) fn build_edge(&self, command: Option<&Command>) -> Option<FragmentEdgeBuildResult> {
+        let (info, replace_job) = match command {
+            None => {
+                return None;
+            }
+            Some(command) => match command {
+                Command::Flush
+                | Command::Pause
+                | Command::Resume
+                | Command::DropStreamingJobs { .. }
+                | Command::MergeSnapshotBackfillStreamingJobs(_)
+                | Command::RescheduleFragment { .. }
+                | Command::SourceChangeSplit(_)
+                | Command::Throttle(_)
+                | Command::CreateSubscription { .. }
+                | Command::DropSubscription { .. } => {
+                    return None;
+                }
+                Command::CreateStreamingJob { info, job_type, .. } => {
+                    let replace_job = match job_type {
+                        CreateStreamingJobType::Normal
+                        | CreateStreamingJobType::SnapshotBackfill(_) => None,
+                        CreateStreamingJobType::SinkIntoTable(replace_job) => Some(replace_job),
+                    };
+                    (Some(info), replace_job)
+                }
+                Command::ReplaceStreamJob(replace_job) => (None, Some(replace_job)),
+            },
+        };
+        let existing_fragment_ids = info
+            .into_iter()
+            .flat_map(|info| info.upstream_fragment_downstreams.keys())
+            .chain(replace_job.into_iter().flat_map(|replace_job| {
+                replace_job
+                    .upstream_fragment_downstreams
+                    .keys()
+                    .filter(|fragment_id| {
+                        info.map(|info| {
+                            !info
+                                .stream_job_fragments
+                                .fragments
+                                .contains_key(fragment_id)
+                        })
+                        .unwrap_or(true)
+                    })
+                    .chain(replace_job.replace_upstream.keys())
+            }))
+            .cloned();
+        let new_fragment_infos = info
+            .into_iter()
+            .flat_map(|info| info.stream_job_fragments.new_fragment_info())
+            .chain(
+                replace_job
+                    .into_iter()
+                    .flat_map(|replace_job| replace_job.new_fragments.new_fragment_info()),
+            )
```

Comment:
This doesn't seem to be readable to me.. Is it possible to somehow reconstruct it?

Reply:
Added a comment to demonstrate how `existing_fragment_ids ` is composed.